### DISCUSSION TITLE
analytics and file_download

### DISCUSSION THREAD

**Discussion by moussatat** [squidfunk/mkdocs-material#7305] [Q&A] [answered]
Hi. didn't seem this question fits with previous ones. I apologize if I should have posted elsewhere.
After adding my G-XXX tag to mkdocs.yml, I also enabled in "enhanced measurements" both search and file download metrics as specified on the documentation site.
In the analytics website, I do see properly the visited page_locations, the search entry made, as well as feedback for pages. But file_download value remains empty when I test it.
I am using a markdown link [```.pdf```](/pdf/path to document.pdf) and host all pdf files in a pdf/ folder in my root directory. 
I tried putting the pdf inside the doc/ folder it wasn't enough.

My question: Is there a specific way to link to a pdf file so that the analytics cookies identify a download when visitors click on the link? Or is there something else at play here.

My mkdocs.yml is [here](https://github.com/moussatat/journal/blob/main/mkdocs.yml)

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @moussatat,
the documentation page doesn't mention anything about file downloads. I'm talking about this page:
- https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/

You can look at the implementation of the built-in analytics code for `search` here:
https://github.com/squidfunk/mkdocs-material/blob/12a8e82837fe400dd1f123e41d75b32987a11744/src/templates/partials/integrations/analytics/google.html#L40-L49

As you can see JavaScript is used to attach a function to the search, which uses the defined `gtag` function to add analytics requests to the stack:
- `gtag("event", "search", { search_term: this.value })`

So you would have to setup a similar code in your extra.js that would attach a special function that would send another "event" using a specified `gtag`:
- https://squidfunk.github.io/mkdocs-material/customization/#additional-javascript

I would use `attr_list` to add a special class to the buttons with download links, to make it easier to target in JavaScript:
- https://squidfunk.github.io/mkdocs-material/reference/buttons/
- `[Download PDF](...){ .md-button .md-button--primary .ext-download-file }`

**Comment by moussatat** [comment]
Thanks for the guidance. I shall generalize the ```attr_list``` for all download clicks.
For the GA4 config, I managed to get a trigger, the report didn't look like standard file_download. I will go without this for now.
